In [11]:
import os
import pandas as pd

# Get your home directory dynamically
home = os.path.expanduser("~")

# Build path to Downloads folder
file_path = os.path.join(home, "Downloads", "sat-results.csv")

# Load the file
df = pd.read_csv(file_path)

print("File loaded successfully:", df.shape)
df.head()


File loaded successfully: (493, 11)


DBN                                    SCHOOL NAME  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2  01M450                     EAST SIDE COMMUNITY SCHOOL   
3  01M458                      FORSYTH SATELLITE ACADEMY   
4  01M509                        MARTA VALLE HIGH SCHOOL   

  Num of SAT Test Takers SAT Critical Reading Avg. Score SAT Math Avg. Score  \
0                     29                             355                 404   
1                     91                             383                 423   
2                     70                             377                 402   
3                      7                             414                 401   
4                     44                             390                 433   

  SAT Writing Avg. Score SAT Critical Readng Avg. Score  internal_school_id  \
0                    363                            355              218160   
1                    366                            383              268547   
2                    370                            377              236446   
3                    359                            414              427826   
4                    384                            390              672714   

  contact_extension pct_students_tested  academic_tier_rating  
0              x345                 78%                   2.0  
1              x234                 NaN                   3.0  
2              x123                 NaN                   3.0  
3              x123                 92%                   4.0  
4              x123                 92%                   2.0

493 rows ,The dataset has 493 entries (schools or records).

11 columns ,Each row has 11 attributes (fields), such as DBN, school name, SAT scores, etc.

In [12]:
print(df.shape)
print(df.columns)
df.head()


(493, 11)
Index(['DBN', 'SCHOOL NAME', 'Num of SAT Test Takers',
       'SAT Critical Reading Avg. Score', 'SAT Math Avg. Score',
       'SAT Writing Avg. Score', 'SAT Critical Readng Avg. Score',
       'internal_school_id', 'contact_extension', 'pct_students_tested',
       'academic_tier_rating'],
      dtype='object')


DBN                                    SCHOOL NAME  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2  01M450                     EAST SIDE COMMUNITY SCHOOL   
3  01M458                      FORSYTH SATELLITE ACADEMY   
4  01M509                        MARTA VALLE HIGH SCHOOL   

  Num of SAT Test Takers SAT Critical Reading Avg. Score SAT Math Avg. Score  \
0                     29                             355                 404   
1                     91                             383                 423   
2                     70                             377                 402   
3                      7                             414                 401   
4                     44                             390                 433   

  SAT Writing Avg. Score SAT Critical Readng Avg. Score  internal_school_id  \
0                    363                            355              218160   
1                    366                            383              268547   
2                    370                            377              236446   
3                    359                            414              427826   
4                    384                            390              672714   

  contact_extension pct_students_tested  academic_tier_rating  
0              x345                 78%                   2.0  
1              x234                 NaN                   3.0  
2              x123                 NaN                   3.0  
3              x123                 92%                   4.0  
4              x123                 92%                   2.0

In [ ]:
CLEAN THE DATA

In [13]:
import pandas as pd

# === 1. Load raw SAT results ===
df = pd.read_csv("sat-results.csv")

print("Original shape:", df.shape)
print("Original columns:", df.columns.tolist())
print(df.head(3))


# === 2. Normalize headers ===
df.columns = (
    df.columns
    .str.strip()
    .str.lower()
    .str.replace(" ", "_")
    .str.replace(r"[^a-z0-9_]", "", regex=True)
)

print("\nNormalized columns:", df.columns.tolist())


# === 3. Remove % signs, convert to numeric ===
for col in df.columns:
    if df[col].dtype == "object":
        # Try cleaning percentages
        df[col] = (
            df[col]
            .astype(str)
            .str.replace("%", "", regex=False)
            .str.replace(",", "", regex=False)
        )

        # Convert to numeric where possible
        df[col] = pd.to_numeric(df[col], errors="ignore")


# === 4. Drop duplicates ===
before = df.shape[0]
df = df.drop_duplicates()
after = df.shape[0]
print(f"\nDropped {before - after} duplicate rows.")


# === 5. Handle SAT score columns (200–800 valid range) ===
sat_cols = [c for c in df.columns if "sat" in c]

for col in sat_cols:
    if pd.api.types.is_numeric_dtype(df[col]):
        df[col] = df[col].where((df[col] >= 200) & (df[col] <= 800))


# === 6. Inspect cleaned data ===
print("\nCleaned data preview:")
print(df.head(5))
print("\nSAT score stats:")
print(df[sat_cols].describe())


# === 7. Save cleaned version ===
df.to_csv("cleaned_sat_results.csv", index=False)
print("\n✅ Cleaned dataset saved as 'cleaned_sat_results.csv'")


Original shape: (493, 11)
Original columns: ['DBN', 'SCHOOL NAME', 'Num of SAT Test Takers', 'SAT Critical Reading Avg. Score', 'SAT Math Avg. Score', 'SAT Writing Avg. Score', 'SAT Critical Readng Avg. Score', 'internal_school_id', 'contact_extension', 'pct_students_tested', 'academic_tier_rating']
      DBN                                    SCHOOL NAME  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2  01M450                     EAST SIDE COMMUNITY SCHOOL   

  Num of SAT Test Takers SAT Critical Reading Avg. Score SAT Math Avg. Score  \
0                     29                             355                 404   
1                     91                             383                 423   
2                     70                             377                 402   

  SAT Writing Avg. Score SAT Critical Readng Avg. Score  internal_school_id  \
0                    363                            355   

/var/folders/pt/44pc3ld561lgr66j_8x0t1vc0000gn/T/ipykernel_50354/52023551.py:35: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df[col] = pd.to_numeric(df[col], errors="ignore")
/var/folders/pt/44pc3ld561lgr66j_8x0t1vc0000gn/T/ipykernel_50354/52023551.py:35: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df[col] = pd.to_numeric(df[col], errors="ignore")
/var/folders/pt/44pc3ld561lgr66j_8x0t1vc0000gn/T/ipykernel_50354/52023551.py:35: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df[col] = pd.to_numeric(df[col], errors="ignore")
/var/folders/pt/44pc3ld561lgr66j_8x0t1vc0000gn/T/ipykernel_50354/52023551.py:35: FutureWarning: errors='ignore' is dep

In [14]:
# === 3. Remove % signs, commas, and convert to numeric where possible ===
for col in df.columns:
    if df[col].dtype == "object":
        # Clean percentages and commas
        df[col] = (
            df[col]
            .astype(str)
            .str.replace("%", "", regex=False)
            .str.replace(",", "", regex=False)
        )

        # Try converting to numeric
        try:
            df[col] = pd.to_numeric(df[col])
        except ValueError:
            # If conversion fails, keep as text
            pass


In [15]:
# Identify SAT score columns (adjust names if needed)
sat_columns = ["sat_math_avg_score", "sat_critical_reading_avg_score", "sat_writing_avg_score"]

for col in sat_columns:
    if col in df.columns:
        # Remove % or commas if present
        df[col] = (
            df[col]
            .astype(str)
            .str.replace("%", "", regex=False)
            .str.replace(",", "", regex=False)
        )
        
        # Convert to numeric
        df[col] = pd.to_numeric(df[col], errors="coerce")


weiter cleaning

In [16]:
import pandas as pd

# Load dataset (update path as needed)
df = pd.read_csv("sat-results.csv")

# Columns for SAT scores
sat_columns = [
    "sat_math_avg_score",
    "sat_critical_reading_avg_score",
    "sat_writing_avg_score"
]

# Clean SAT columns
for col in sat_columns:
    if col in df.columns:
        df[col] = (
            df[col]
            .astype(str)
            .str.replace("%", "", regex=False)
            .str.replace(",", "", regex=False)
        )
        df[col] = pd.to_numeric(df[col], errors="coerce")

        # Keep only valid SAT scores (200–800)
        df.loc[(df[col] < 200) | (df[col] > 800), col] = None

# Drop rows wit


In [1]:
schools_per_borough = df_clean.groupby("borough")["dbn"].nunique().reset_index()
print(schools_per_borough)


NameError: name 'df_clean' is not defined

In [18]:
import pandas as pd

# Load dataset
df = pd.read_csv("sat-results.csv")

# Columns for SAT scores
sat_columns = [
    "sat_math_avg_score",
    "sat_critical_reading_avg_score",
    "sat_writing_avg_score"
]

# Clean SAT columns
for col in sat_columns:
    if col in df.columns:
        df[col] = (
            df[col]
            .astype(str)
            .str.replace("%", "", regex=False)
            .str.replace(",", "", regex=False)
        )
        df[col] = pd.to_numeric(df[col], errors="coerce")

        # Keep only valid SAT scores (200–800)
        df.loc[(df[col] < 200) | (df[col] > 800), col] = None

# Drop rows with


In [24]:
print(df.columns.tolist())


['DBN', 'SCHOOL NAME', 'Num of SAT Test Takers', 'SAT Critical Reading Avg. Score', 'SAT Math Avg. Score', 'SAT Writing Avg. Score', 'SAT Critical Readng Avg. Score', 'internal_school_id', 'contact_extension', 'pct_students_tested', 'academic_tier_rating']


In [25]:
import pandas as pd

# Load your dataset
df = pd.read_csv("sat-results.csv")

# Correct SAT score columns based on your file
sat_columns = [
    "SAT Critical Reading Avg. Score",
    "SAT Math Avg. Score",
    "SAT Writing Avg. Score"
]

# Clean SAT columns
for col in sat_columns:
    if col in df.columns:
        df[col] = (
            df[col]
            .astype(str)
            .str.replace("%", "", regex=False)   # remove % if any
            .str.replace(",", "", regex=False)   # remove commas
        )
        df[col] = pd.to_numeric(df[col], errors="coerce")

        # Keep only valid SAT scores (200–800)
        df.loc[(df[col] < 200) | (df[col] > 800), col] = None

# Drop rows where all SAT scores are missing
df_clean = df.dropna(subset=sat_columns, how="all")

print("✅ Data cleaned. Shape:", df_clean.shape)
df_clean.head()


✅ Data cleaned. Shape: (435, 11)


DBN                                    SCHOOL NAME  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2  01M450                     EAST SIDE COMMUNITY SCHOOL   
3  01M458                      FORSYTH SATELLITE ACADEMY   
4  01M509                        MARTA VALLE HIGH SCHOOL   

  Num of SAT Test Takers  SAT Critical Reading Avg. Score  \
0                     29                            355.0   
1                     91                            383.0   
2                     70                            377.0   
3                      7                            414.0   
4                     44                            390.0   

   SAT Math Avg. Score  SAT Writing Avg. Score SAT Critical Readng Avg. Score  \
0                404.0                   363.0                            355   
1                423.0                   366.0                            383   
2                402.0                   370.0                            377   
3                401.0                   359.0                            414   
4                433.0                   384.0                            390   

   internal_school_id contact_extension pct_students_tested  \
0              218160              x345                 78%   
1              268547              x234                 NaN   
2              236446              x123                 NaN   
3              427826              x123                 92%   
4              672714              x123                 92%   

   academic_tier_rating  
0                   2.0  
1                   3.0  
2                   3.0  
3                   4.0  
4                   2.0

In [ ]:
CHECK COLUMN NAMES

In [26]:
# Normalize column names
df.columns = (
    df.columns
    .str.strip()
    .str.lower()
    .str.replace(" ", "_")
    .str.replace(r"[^a-z0-9_]", "", regex=True)  # remove non-alphanumeric chars
)

# Check new column names
print(df.columns.tolist())


['dbn', 'school_name', 'num_of_sat_test_takers', 'sat_critical_reading_avg_score', 'sat_math_avg_score', 'sat_writing_avg_score', 'sat_critical_readng_avg_score', 'internal_school_id', 'contact_extension', 'pct_students_tested', 'academic_tier_rating']


In [27]:
sat_columns = [
    "sat_critical_reading_avg_score",
    "sat_math_avg_score",
    "sat_writing_avg_score"
]

for col in sat_columns:
    if col in df.columns:
        df[col] = pd.to_numeric(df[col], errors="coerce")
        df.loc[(df[col] < 200) | (df[col] > 800), col] = None

# Drop rows where all SAT scores are missing
df_clean = df.dropna(subset=sat_columns, how="all")

print("✅ Data cleaned. Shape:", df_clean.shape)
df_clean.head()


✅ Data cleaned. Shape: (435, 11)


dbn                                    school_name  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2  01M450                     EAST SIDE COMMUNITY SCHOOL   
3  01M458                      FORSYTH SATELLITE ACADEMY   
4  01M509                        MARTA VALLE HIGH SCHOOL   

  num_of_sat_test_takers  sat_critical_reading_avg_score  sat_math_avg_score  \
0                     29                           355.0               404.0   
1                     91                           383.0               423.0   
2                     70                           377.0               402.0   
3                      7                           414.0               401.0   
4                     44                           390.0               433.0   

   sat_writing_avg_score sat_critical_readng_avg_score  internal_school_id  \
0                  363.0                           355              218160   
1                  366.0                           383              268547   
2                  370.0                           377              236446   
3                  359.0                           414              427826   
4                  384.0                           390              672714   

  contact_extension pct_students_tested  academic_tier_rating  
0              x345                 78%                   2.0  
1              x234                 NaN                   3.0  
2              x123                 NaN                   3.0  
3              x123                 92%                   4.0  
4              x123                 92%                   2.0

In [28]:
df_clean.to_csv("cleaned_sat_results.csv", index=False)


In [29]:
import pandas as pd

# Assuming df_clean is ready from the cleaning step
# Add a total SAT score column
df_clean["total_sat"] = (
    df_clean["sat_critical_reading_avg_score"] +
    df_clean["sat_math_avg_score"] +
    df_clean["sat_writing_avg_score"]
)

# 1. Schools with valid SAT scores
valid_schools = df_clean["dbn"].nunique()
print("✅ Schools with valid SAT scores:", valid_schools)

# 2. Average SAT scores
avg_scores = df_clean[["sat_critical_reading_avg_score",
                       "sat_math_avg_score",
                       "sat_writing_avg_score",
                       "total_sat"]].mean().round(2)
print("\n📊 Average SAT Scores:")
print(avg_scores)

# 3. Top 10 schools by total SAT
top10 = df_clean.nlargest(10, "total_sat")[["school_name", "total_sat"]]
print("\n🏆 Top 10 Schools by Total SAT:")
print(top10)

# 4. Worst 10 schools by total SAT
bottom10 = df_clean.nsmallest(10, "total_sat")[["school_name", "total_sat"]]
print("\n⚠️ Bottom 10 Schools by Total SAT:")
print(bottom10)

# 5. Distribution summary for Math
print("\n📈 Math Score Distribution:")
print(df_clean["sat_math_avg_score"].describe())


✅ Schools with valid SAT scores: 421

📊 Average SAT Scores:
sat_critical_reading_avg_score     400.71
sat_math_avg_score                 413.34
sat_writing_avg_score              393.71
total_sat                         1208.16
dtype: float64

🏆 Top 10 Schools by Total SAT:
                                           school_name  total_sat
48                              STUYVESANT HIGH SCHOOL     2096.0
198                       BRONX HIGH SCHOOL OF SCIENCE     1969.0
459                STATEN ISLAND TECHNICAL HIGH SCHOOL     1953.0
206  HIGH SCHOOL OF AMERICAN STUDIES AT LEHMAN COLLEGE     1920.0
396                        TOWNSEND HARRIS HIGH SCHOOL     1910.0
427  QUEENS HIGH SCHOOL FOR THE SCIENCES AT YORK CO...     1868.0
8                       BARD HIGH SCHOOL EARLY COLLEGE     1856.0
107  HIGH SCHOOL FOR MATHEMATICS, SCIENCE AND ENGIN...     1847.0
249                     BROOKLYN TECHNICAL HIGH SCHOOL     1833.0
33                       ELEANOR ROOSEVELT HIGH SCHOOL     1758.0

/var/folders/pt/44pc3ld561lgr66j_8x0t1vc0000gn/T/ipykernel_50354/1224459323.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean["total_sat"] = (


CHECK IF YOUR CLEAN DATA EXIST

In [31]:
import pandas as pd
df=pd.read_csv('sat-results.csv')
df

DBN                                        SCHOOL NAME  \
0    01M292      HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1    01M448                UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2    01M450                         EAST SIDE COMMUNITY SCHOOL   
3    01M458                          FORSYTH SATELLITE ACADEMY   
4    01M509                            MARTA VALLE HIGH SCHOOL   
..      ...                                                ...   
488  27Q480                             JOHN ADAMS HIGH SCHOOL   
489  13K605  GEORGE WESTINGHOUSE CAREER AND TECHNICAL EDUCA...   
490  05M304                              MOTT HALL HIGH SCHOOL   
491  02M520   MURRY BERGTRAUM HIGH SCHOOL FOR BUSINESS CAREERS   
492  07X221    SOUTH BRONX PREPARATORY: A COLLEGE BOARD SCHOOL   

    Num of SAT Test Takers SAT Critical Reading Avg. Score  \
0                       29                             355   
1                       91                             383   
2                       70                             377   
3                        7                             414   
4                       44                             390   
..                     ...                             ...   
488                    403                             391   
489                     85                             406   
490                     54                             413   
491                    264                             407   
492                     65                             364   

    SAT Math Avg. Score SAT Writing Avg. Score SAT Critical Readng Avg. Score  \
0                   404                    363                            355   
1                   423                    366                            383   
2                   402                    370                            377   
3                   401                    359                            414   
4                   433                    384                            390   
..                  ...                    ...                            ...   
488                 409                    392                            391   
489                 391                    392                            406   
490                 399                    398                            413   
491                 440                    393                            407   
492                 378                    348                            364   

     internal_school_id contact_extension pct_students_tested  \
0                218160              x345                 78%   
1                268547              x234                 NaN   
2                236446              x123                 NaN   
3                427826              x123                 92%   
4                672714              x123                 92%   
..                  ...               ...                 ...   
488              863765               NaN                 92%   
489              937579              x234                 NaN   
490              296405              x123                 78%   
491              892839               NaN                 92%   
492              277389              x345                 92%   

     academic_tier_rating  
0                     2.0  
1                     3.0  
2                     3.0  
3                     4.0  
4                     2.0  
..                    ...  
488                   1.0  
489                   NaN  
490                   2.0  
491                   2.0  
492                   NaN  

[493 rows x 11 columns]

In [ ]:
CHECK DUPLICATES

In [32]:
new_df=df.drop_duplicates()
new_df

DBN                                    SCHOOL NAME  \
0    01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1    01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2    01M450                     EAST SIDE COMMUNITY SCHOOL   
3    01M458                      FORSYTH SATELLITE ACADEMY   
4    01M509                        MARTA VALLE HIGH SCHOOL   
..      ...                                            ...   
473  75X012               P.S. X012 LEWIS AND CLARK SCHOOL   
474  75X754        J. M. RAPPORT SCHOOL CAREER DEVELOPMENT   
475  79M645     SCHOOL FOR COOPERATIVE TECHNICAL EDUCATION   
476  79Q950                            GED PLUS s CITYWIDE   
477  79X490                                PHOENIX ACADEMY   

    Num of SAT Test Takers SAT Critical Reading Avg. Score  \
0                       29                             355   
1                       91                             383   
2                       70                             377   
3                        7                             414   
4                       44                             390   
..                     ...                             ...   
473                      s                               s   
474                      s                               s   
475                      s                               s   
476                      8                             496   
477                      9                             367   

    SAT Math Avg. Score SAT Writing Avg. Score SAT Critical Readng Avg. Score  \
0                   404                    363                            355   
1                   423                    366                            383   
2                   402                    370                            377   
3                   401                    359                            414   
4                   433                    384                            390   
..                  ...                    ...                            ...   
473                   s                      s                              s   
474                   s                      s                              s   
475                   s                      s                              s   
476                 400                    426                            496   
477                 370                    360                            367   

     internal_school_id contact_extension pct_students_tested  \
0                218160              x345                 78%   
1                268547              x234                 NaN   
2                236446              x123                 NaN   
3                427826              x123                 92%   
4                672714              x123                 92%   
..                  ...               ...                 ...   
473              733698              x234                 92%   
474              976034              x345                 NaN   
475              556924              x123                 85%   
476              544514              x234                 92%   
477              248647              x234                 85%   

     academic_tier_rating  
0                     2.0  
1                     3.0  
2                     3.0  
3                     4.0  
4                     2.0  
..                    ...  
473                   1.0  
474                   4.0  
475                   1.0  
476                   2.0  
477                   NaN  

[478 rows x 11 columns]

In [ ]:
CONNECT WITH NEON

In [35]:
from sqlalchemy import create_engine

engine = create_engine(
    "postgresql+psycopg2://username:password@localhost:5432/dbname"
)


In [39]:
# SQLAlchemy connection string format:
# postgresql+psycopg2://user:password@host:port/dbname

DATABASE_URL = (
    "postgresql+psycopg2://neondb_owner:npg_CeS9fJg2azZD"
    "@ep-falling-glitter-a5m0j5gk-pooler.us-east-2.aws.neon.tech:5432/neondb"
    "?sslmode=require"
)

# Create engine and establish connection
engine = create_engine(DATABASE_URL)

In [40]:
new_df.to_sql(
    name='vanessa_sat_results',       
    con=engine,     
    schema='nyc_schools',
    if_exists='replace',    
    index=False            
)

478

In [42]:
from sqlalchemy import text

with engine.connect() as conn:
    result = conn.execute(text("SELECT current_database();"))
    print("✅ Connecté à la base :", result.scalar())


✅ Connecté à la base : neondb


In [43]:
from sqlalchemy import inspect

# Create an inspector object
inspector = inspect(engine)

# List all schemas
schemas = inspector.get_schema_names()
print("📂 Available schemas:", schemas)

# List tables in your schema (replace 'nyc_schools' if needed)
tables = inspector.get_table_names(schema="nyc_schools")
print("📑 Tables in nyc_schools schema:", tables)


📂 Available schemas: ['dependency_example', 'information_schema', 'nyc_schools', 'public', 'test_berlin_data']
📑 Tables in nyc_schools schema: ['school_safety_report', 'high_school_directory', 'serhii_sotnichenko_sat_scores', 'mytable', 'school_demographics', 'sat_scores_cleaned', 'yana_yelnikova_sat_scores', 'meifang_sat_results', 'heike_reichert_sat_results', 'anastasia_sat_results', 'vanessa_sat_results', 'cleaned_sat_results']


In [44]:
import pandas as pd

query = "SELECT * FROM nyc_schools.vanessa_sat_results LIMIT 5;"
df_preview = pd.read_sql(query, engine)

df_preview


DBN                                    SCHOOL NAME  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2  01M450                     EAST SIDE COMMUNITY SCHOOL   
3  01M458                      FORSYTH SATELLITE ACADEMY   
4  01M509                        MARTA VALLE HIGH SCHOOL   

  Num of SAT Test Takers SAT Critical Reading Avg. Score SAT Math Avg. Score  \
0                     29                             355                 404   
1                     91                             383                 423   
2                     70                             377                 402   
3                      7                             414                 401   
4                     44                             390                 433   

  SAT Writing Avg. Score SAT Critical Readng Avg. Score  internal_school_id  \
0                    363                            355              218160   
1                    366                            383              268547   
2                    370                            377              236446   
3                    359                            414              427826   
4                    384                            390              672714   

  contact_extension pct_students_tested  academic_tier_rating  
0              x345                 78%                   2.0  
1              x234                None                   3.0  
2              x123                None                   3.0  
3              x123                 92%                   4.0  
4              x123                 92%                   2.0

In [45]:
new_df.to_csv("cleaned_sat_results.csv", index=False)


In [3]:
import pandas as pd

# Load the cleaned CSV instead of the raw one
df = pd.read_csv("cleaned_sat_results.csv")

print("✅ Data ready for visualization:", df.shape)


✅ Data ready for visualization: (478, 11)


In [5]:
import matplotlib.pyplot as plt

plt.figure(figsize=(10,6))
df[['sat_math_avg_score', 'sat_critical_reading_avg_score', 'sat_writing_avg_score']].hist(
    bins=20, figsize=(12,6), layout=(1,3), edgecolor='black'
)
plt.suptitle("Distribution of SAT Scores Across NYC Schools", fontsize=16)
plt.show()



KeyError: "None of [Index(['sat_math_avg_score', 'sat_critical_reading_avg_score',\n       'sat_writing_avg_score'],\n      dtype='object')] are in the [columns]"

<Figure size 1000x600 with 0 Axes>

In [6]:
# Rename columns for consistency
df = df.rename(columns={
    "SAT Math Avg. Score": "sat_math_avg_score",
    "SAT Critical Reading Avg. Score": "sat_critical_reading_avg_score",
    "SAT Writing Avg. Score": "sat_writing_avg_score"
})


In [7]:
import matplotlib.pyplot as plt

plt.figure(figsize=(10,6))
df[['sat_math_avg_score', 'sat_critical_reading_avg_score', 'sat_writing_avg_score']].hist(
    bins=20, figsize=(12,6), layout=(1,3), edgecolor='black'
)
plt.suptitle("Distribution of SAT Scores Across NYC Schools", fontsize=16)
plt.show()


ValueError: hist method requires numerical or datetime columns, nothing to plot.

<Figure size 1000x600 with 0 Axes>